src/codewhisperer/util/dependencyGraph/goDependencyGraph.ts

```diff
@@ -0,0 +1,236 @@
+    override async getSourceDependencies(uri: Uri, content: string): Promise<string[]> {
+        const imports = this.readImports(content)
+        const dependencies = this.getDependencies(uri, imports)
+        return dependencies
+    }
+
+    // Returns file paths of other .go files in the same directory declared with the same package statement.
+    override async getSamePackageFiles(uri: Uri): Promise<string[]> {
+        const fileList: string[] = []
+        const packagePath = path.dirname(uri.fsPath)
+        const fileName = path.basename(uri.fsPath)
+        const content = await readFileAsString(uri.fsPath)
+        const packageName = this.readPackageName(content)
+
+        const files = readdirSync(packagePath, { withFileTypes: true })
+        for (const file of files) {
+            if (file.isDirectory() || !file.name.endsWith(DependencyGraphConstants.goExt) || file.name === fileName) {
+                continue
+            }
+            const filePath = path.join(packagePath, file.name)
+            const content = await readFileAsString(filePath)
+            if (this.readPackageName(content) !== packageName) {
+                continue
+            }
+            fileList.push(filePath)
+        }
+
+        return fileList
+    }
+
+    override async isTestFile(content: string): Promise<boolean> {
+        const imports = this.readImports(content)
+        const filteredImports = imports.filter(importStr => {
+            return importStr.includes('"testing"')
+        })
+        return filteredImports.length > 0
+    }
+
+    override async generateTruncation(uri: Uri): Promise<Truncation> {
+        try {
+            const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+            if (workspaceFolder === undefined) {
+                this._pickedSourceFiles.add(uri.fsPath)
+            } else {
+                await this.searchDependency(uri)
+                await this.traverseDir(this.getProjectPath(uri))
+            }
+            await sleep(1000)
+            getLogger().verbose('Picked source files: [' + [...this._pickedSourceFiles].join(', ') + ']')
+            const truncDirPath = this.getTruncDirPath(uri)
+            this.copyFilesToTmpDir(this._pickedSourceFiles, truncDirPath)
+            const zipFilePath = this.zipDir(truncDirPath, CodeWhispererConstants.codeScanZipExt)
+            const zipFileSize = statSync(zipFilePath).size
+            return {
+                rootDir: truncDirPath,
+                zipFilePath,
+                scannedFiles: new Set(this._pickedSourceFiles),
+                srcPayloadSizeInBytes: this._totalSize,
+                zipFileSizeInBytes: zipFileSize,
+                buildPayloadSizeInBytes: 0,
+                lines: this._totalLines,
+            }
+        } catch (error) {
+            getLogger().error('Go dependency graph error caused by:', error)
+            throw new Error('Go context processing failed.')
```

Comment:
Should we throw a toolkit error instead?
```suggestion
            throw ToolkitError.chain(error, 'Go context processing failed.')
```